src/common/hobject.h

```diff
@@ -445,6 +445,18 @@ struct ghobject_t {
     (*this) = temp;
   }
 
+  operator std::string() {
+    std::stringstream out;
+    out << this;
```

Comment:
It looks like this will write the address of "this" to the stream, rather than the value, *this. Is that what you want? Is there some other magic happening here that writes the value instead? Do I just need coffee? :-)

Reply:
That is correct, it's for the implicit cast to string for unknown types. A simple `str()` method would also work, and we'd append the `.str()` call in tracetool.